rq/job.py

```diff
@@ -485,6 +485,8 @@ def delete(self, pipeline=None):
     # Job execution
     def perform(self):  # noqa
         """Invokes the job function with the job arguments."""
+        self.connection.persist(self.key)
+        self.ttl = self.connection.ttl(self.key)
```

Comment:
Ah, I missed your comment above. Yes, please hard code it to -1. This saves a round trip to Redis :)